### DISCUSSION TITLE
Tensorflow & other binaries downgrade during install.

### DISCUSSION THREAD

**Discussion by MicahNjeru** [mindee/doctr#965] [General]
When installing docTR (great model btw), it does this "weird" thing of uninstalling a couple of my modules such as numpy, tensorflow & h5py etc then installing them. Which is okay-ish but it leaves them with an older version and specifically for Tensorflow. So when you run the model it tells you you're running on an older version of tensorflow yet it's the installation that downgraded it.
Also, why does it have to download multiple tensorflow version beginning with the latest then downloading the older versions as it installs
![Screenshot (1)](https://user-images.githubusercontent.com/97245548/176124727-d28c2679-ba47-437b-86e5-3489172aea00.png)
![Screenshot (3)](https://user-images.githubusercontent.com/97245548/176124769-40ea99ae-5524-40c2-a905-ffa3fcfa99a7.png)
?

**Comment by felixdittrich92** [comment]
Hi @MicahNjeru,

Yes that's true there was a backward compatibility issue beetwen TF and Keras (since Keras 2.7 release) #579, so downgrading was a temporally fix.

This is fixed on main branch since #963 is merged :+1: 
We will release it with version 0.5.2 soon :smiley: 
Additional #957 will also improve the dependency management

**Reply by MicahNjeru** [reply]
Can't wait for the update.

**Reply by felixdittrich92** [reply]
#967 :+1:
